src/main/java/com/artipie/RepoConfig.java

```diff
@@ -88,6 +88,14 @@ public RepoConfig(final Publisher<ByteBuffer> content) {
             .thenApply(YamlStorageSettings::storage);
     }
 
+    /**
+     * Custom repository configuration.
+     * @return Async custom repository config
+     */
+    public CompletionStage<YamlMapping> custom() {
```

Comment:
@brastak we should add  a non null check `Objects.requireNonNull` like we have for `repo()` method

Reply:
@Vatavuk I did the other thing - I have changed signature to return Optional, because of these section is not required